File: iosSDK/text/XML/XML/ViewController.m
//
//  ViewController.m
//  XML
//
//  Created by  wj on 16/5/5.
//  Copyright © 2016年 gzq. All rights reserved.
//

#import "ViewController.h"
#import "GDataXMLNode.h"

@interface ViewController (){
    
    GDataXMLDocument *_document;
}

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    
    NSString *path = [[NSBundle mainBundle]pathForResource:@"xml" ofType:@"txt"];
    
    NSData *data = [NSData dataWithContentsOfFile:path];
    
    _document = [[GDataXMLDocument alloc]initWithData:data options:0 error:nil];
    
    GDataXMLElement *rootElement = [_document rootElement];
    
    NSArray *booksElementArray = [rootElement elementsForName:@"books"];
    
    GDataXMLElement *booksElement = booksElementArray[0];
    
    NSArray *bookElementArray = [booksElement elementsForName:@"book"];
    
    for(GDataXMLElement *bookElement in booksElementArray){
        
        NSArray *summaryElementArray = [bookElement elementsForName:@"summary"];
        
        GDataXMLElement *summaryElement = summaryElementArray[0];
        
        NSLog(@"%@",[summaryElement stringValue]);
    }
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: iosSDK/SQLitePersistence/SQLitePersistence/ViewController.swift
//
//  ViewController.swift
//  SQLitePersistence
//
//  Created by  wj on 16/1/6.
//  Copyright © 2016年 gzq. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    @IBOutlet var lineFields: [UITextField]!
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        
        var database:COpaquePointer = nil
        var result = sqlite3_open(dataFilePath(),database)
        != SQLITE_OK{
            sqlite3_close(database)
            print("失败打开数据库")
            return
        }
        
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    func dataFilePath() -> String{
        let paths = NSSearchPathForDirectoriesInDomains(.DocumentDirectory, .UserDomainMask, true)
        let documentsDirectory = paths[0] as NSString
        return documentsDirectory.stringByAppendingPathComponent("data.sqlite") as String
    }
}

